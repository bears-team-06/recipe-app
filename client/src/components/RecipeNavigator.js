import React from 'react'
import styled from 'styled-components'

const RecipeList = ({ recipes, onClick }) => {
    if (recipes && Array.isArray(recipes)) {
        return recipes.map((recipe,index) => <StyledRecipeListItem className={`recipeItem-${index}`} key={recipe.replace(" ","")} onClick={() => onClick(index)}>{recipe}</StyledRecipeListItem>);
    }
}

const StyledRecipeListItem = styled.li`
  font-size: 20px;
  list-style: none;
  padding: 5px;
  white-space: nowrap;
`

const RecipeNavigator = (props) => {
    return (
        <div>
        <List>
            <RecipeList recipes={props.recipes} onClick={props.onClick}/>
        </List>
            {props.children}
        </div>
    )
}

const List = styled.ul`
  width: auto;
  margin: 10px;
`



export default RecipeNavigator;

