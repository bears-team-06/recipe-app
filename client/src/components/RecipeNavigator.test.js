import React from "react";
import RecipeNavigator from "./RecipeNavigator";
import renderer from 'react-test-renderer';

describe("Recipe Navigator", () => {
    it('matches snapshot', () => {
        const RecipeNavigatorWrapper = renderer.create(<RecipeNavigator recipes={["recipe1","recipe2"]}/>).toJSON()
        expect(RecipeNavigatorWrapper).toMatchSnapshot()
    })

    it('returns index of recipe that was clicked', () => {
        expect(true).toBe(true)
    })
});
