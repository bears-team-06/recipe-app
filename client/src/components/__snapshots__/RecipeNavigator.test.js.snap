// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe Navigator matches snapshot 1`] = `
.c1 {
  font-size: 20px;
  list-style: none;
  padding: 5px;
  white-space: nowrap;
}

.c0 {
  width: auto;
  margin: 10px;
}

<div>
  <ul
    className="c0"
  >
    <li
      className="recipeItem-0 c1"
      onClick={[Function]}
    >
      recipe1
    </li>
    <li
      className="recipeItem-1 c1"
      onClick={[Function]}
    >
      recipe2
    </li>
  </ul>
</div>
`;

exports[`Recipe Navigator matches snapshot when child elements are passed 1`] = `
.c1 {
  font-size: 20px;
  list-style: none;
  padding: 5px;
  white-space: nowrap;
}

.c0 {
  width: auto;
  margin: 10px;
}

<div>
  <ul
    className="c0"
  >
    <li
      className="recipeItem-0 c1"
      onClick={[Function]}
    >
      recipe1
    </li>
    <li
      className="recipeItem-1 c1"
      onClick={[Function]}
    >
      recipe2
    </li>
  </ul>
  <h1>
    child element
  </h1>
</div>
`;
